src/components/ebay-calendar/component.js

```diff
@@ -0,0 +1,383 @@
+        }
+    }
+
+    /**
+     * @param {DayISO} iso
+     */
+    isDisabled(iso) {
+        return (
+            (this.state.disableBefore && iso < this.state.disableBefore) ||
+            (this.state.disableAfter && iso > this.state.disableAfter) ||
+            this.state.disableWeekdays.includes(new Date(iso).getUTCDay()) ||
+            this.state.disableList.includes(iso)
+        );
+    }
+
+    /**
+     * @param {DayISO} time
+     */
+    onDaySelect(time) {
+        this.emit('select', { iso: time });
+    }
+
+    /**
+     * @param {DayISO} time
+     */
+    onDayFocus(time) {
+        this.state.focusISO = this.state.tabindexISO = time;
+        this.calculateRangeDisplay();
+    }
+
+    onDayBlur() {
+        this.state.focusISO = undefined;
+        this.calculateRangeDisplay();
+    }
+
+    /**
+     * @param {KeyboardEvent} event
+     */
+    onKeyDown(event) {
+        const dayChange = DAY_UPDATE_KEYMAP[event.key];
+        if (dayChange) {
+            event.preventDefault();
+            // find new tabindex iso, skipping up to 7 disabled cells
+            let curr = fromISO(this.state.tabindexISO);
+            let tries = 7;
+            let iso;
+            do {
+                const next = new Date(
+                    curr.getFullYear(),
+                    curr.getMonth(),
+                    curr.getDate() + dayChange
+                );
+                iso = toISO(next);
+                curr = next;
+            } while (tries-- > 0 && this.isDisabled(iso));
+            if (tries > 0) {
+                // check for edges of calendar
+                const firstVisibleISO = this.getFirstVisibleISO();
+                const lastVisibleISO = this.getLastVisibleISO();
+                if (iso < firstVisibleISO) {
+                    if (this.input.navigable) this.prevMonth();
+                    else iso = firstVisibleISO;
+                } else if (iso > lastVisibleISO) {
+                    if (this.input.navigable) this.nextMonth();
+                    else iso = lastVisibleISO;
+                }
+                this.setTabindexAndFocus(iso);
+            }
+        } else {
+            switch (event.key) {
+                case 'PageUp':
+                    this.prevMonth(true);
+                    break;
+                case 'PageDown':
+                    this.nextMonth(true);
+                    break;
+                case 'Home':
+                    this.setTabindexAndFocus(this.getFirstActiveISO());
+                    break;
+                case 'End':
+                    this.setTabindexAndFocus(this.getLastActiveISO());
+                    break;
+                default:
+            }
+        }
+        this.emit('keydown', event);
+    }
+
+    /**
+     * @param {number} offset
+     */
+    getMonthDate(offset) {
+        const baseDate = fromISO(this.state.baseISO);
+        return new Date(baseDate.getFullYear(), baseDate.getMonth() + offset);
```

Comment:
I see this kind of `new Date()` method being used a lot in similar ways. Maybe we can have a helper function which calculates what we need
Like
`generateDate(baseDate, {month: 1})` which increments month by 1,
`generateDate(baseDate, {day: 1})` which increments day by 1,
etc? Thoughts?